### PR TITLE
[Lee Chun Jie] iP

### PR DESCRIPTION
**Pull request by CJ-Lee01**
# duke.Duke

>You fail to plan you plan to fail - [I don't know who said it](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

- Set Deadlines
- Set ToDos
- Set Events
- Mark and Unmark Tasks.

## What to do?
1. Download it
2. Run it
3. Add your tasks
4. ~~Profit~~ 💹

## Some `code`
```java
class Main {
    public static void main(String[] args) {
        // run Duke.
    }
}
```

## Some checklist
- [ ] bar
- [ ] foo

### CODE REVIEW

File: src/main/java/duke/Deadline.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        super(item);
+        this.deadline = deadline;
+    }
+    /**
+     * Gives a string representation of the task in file format.
+     *
+     * @return A string representation of the task in file format.
+     */
+    @Override
+    public String fileString() {
```
Comment (AprupKale) [inline src/main/java/duke/Deadline.java]: Wouldn't getFileString be a better name than fileString for this method?

File: src/main/java/duke/Duke.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package duke;
+
+/**
+ * This represents the main entry point of the Duke program.
+ */
+public class Duke {
+    /** The name of the bot. */
+    private static final String NAME = "Nino!";
+    /**
+     * runs the Duke program.
```
Comment (AprupKale) [inline src/main/java/duke/Duke.java:36]: Nitpick: Runs instead of runs

File: src/main/java/duke/Parser.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+     * @throws DukeException If the array of strings does not contain enough information to create a Task object.
+     */
+    public static Task parseTask(Commands cmd, String[] instructions) throws DukeException {
+        if (instructions.length != 2) {
+            throw new DukeException("No details given for command: " + cmd);
+        }
+        return Task.create(cmd, instructions[1]);
+    }
+
+    public static String findAll(String[] instructions, StoreList src) {
```
Comment (AprupKale) [inline src/main/java/duke/Parser.java:107]: Not sure findAll is totally descriptive. findAllTasks seems more descriptive.

File: src/main/java/duke/StoreList.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+    /**
+     * Adds task corresponding to the Commands enum value and a string representing the description of the task.
+     * If a task cannot be created from the given information, it returns an error message.
+     *
+     * @param type The Commands enum value representing the type of task to be added.
+     * @param description The description of the task to be added.
+     * @return A string representing the result of adding the task to the list or an error if unsuccessful.
+     */
+    String add(Commands type, String description) {
```
Comment (AprupKale) [inline src/main/java/duke/StoreList.java]: Missing access specifier!

File: src/main/java/duke/StoreList.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+    /**
+     * Marks the task at the string representation of the position of the task.
+     * If the position is not a valid integer, it returns an error message.
+     * If the position is out of bounds, it returns an error message.
+     *
+     * @param position The position of the task in the list as a string.
+     * @return A string representing the result of marking the task as done.
+     */
+    String markDone(String position) {
```
Comment (AprupKale) [inline src/main/java/duke/StoreList.java]: Missing access specifier!

File: src/main/java/duke/StoreList.java [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+    /**
+     * Unmarks the task at the string representation of the position of the task.
+     * If the position is not a valid integer, it returns an error message.
+     * If the position is out of bounds, it returns an error message.
+     *
+     * @param position The position of the task in the list as a string.
+     * @return A string representing the result of marking the task as not done.
+     */
+    String markUndone(String position) {
```
Comment (AprupKale) [inline src/main/java/duke/StoreList.java]: Missing access specifier!

File: src/main/java/duke/StoreList.java [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+    /**
+     * Deletes task based on the position from parsing the string.
+     * If the position is not a valid integer, it returns an error message.
+     * If the position is out of bounds, it returns an error message.
+     *
+     * @param position The position of the task in the list as a string.
+     * @return A string representing the result of removing the task from the list.
+     */
+    String delete(String position) {
```
Comment (AprupKale) [inline src/main/java/duke/StoreList.java]: Missing access specifier!

File: src/main/java/duke/UserInterface.java [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    /**
+     * Displays a string to a user in a formatted manner.
+     *
+     * @param s The string to be displayed to the user.
+     */
+    public static void display(String s) {
+        System.out.println(wrapper(s));
+    }
+
+    private static String wrapper(String line) {
```
Comment (AprupKale) [inline src/main/java/duke/UserInterface.java]: Can displayWrapper be a better name than wrapper?

File: src/test/java/duke/DeadlineTest.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class DeadlineTest {
+
+    @Test
+    public void cannotCreateDeadlineWithoutDateTime() {
```
Comment (AprupKale) [inline src/test/java/duke/DeadlineTest.java]: create_withouthDateTime_throwsException follows the convention

File: src/test/java/duke/DeadlineTest.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        assertThrows(DukeException.class, () -> {
+            Deadline deadline = Deadline.create("test");
+        });
+    }
+
+    @Test
+    public void cannotCreateEmptyDeadline() {
```
Comment (AprupKale) [inline src/test/java/duke/DeadlineTest.java]: create_emptyDeadline_throwsException follows the convention

File: src/test/java/duke/TodoTest.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TodoTest {
+
+    @Test
+    public void cannotCreateEmptyTodo() {
```
Comment (AprupKale) [inline src/test/java/duke/TodoTest.java]: create_emptyTodo_throwsException follows the convention

File: src/test/java/duke/EventTest.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class EventTest {
+
+    @Test
+    public void cannotCreateEventWithoutDateTime() {
```
Comment (AprupKale) [inline src/test/java/duke/EventTest.java]: create_withouthDateTime_throwsException follows the convention

File: src/test/java/duke/EventTest.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+        assertThrows(DukeException.class, () -> {
+            Event event = Event.create("event");
+        });
+    }
+
+    @Test
+    public void cannotCreateEmptyEvent() {
```
Comment (AprupKale) [inline src/test/java/duke/EventTest.java]: create_emptyEvent_throwsException follows the convention

**Review by AprupKale — COMMENTED** [review]
LGTM! Very good code structure overall. Just minor issues and one nitpick to change.

File: src/test/java/duke/EventTest.java (hunk 13 above)
Comment (CJ-Lee01) [reply src/test/java/duke/EventTest.java]: Thank you for the feedback.

**Pull request closed without merging**